conda/cli/main_compare.py

```diff
@@ -130,11 +130,10 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
         raise
 
     active_pkgs = {pkg.name: pkg for pkg in get_packages(prefix)}
-    specification_pkgs = []
-    if "conda" in env.dependencies:
-        specification_pkgs = specification_pkgs + env.dependencies["conda"]
-    if "pip" in env.dependencies:
-        specification_pkgs = specification_pkgs + env.dependencies["pip"]
+    specification_pkgs = env.requested_packages
+    if len(env.external_packages) > 0:
+        for package_type, packages in env.external_packages.items():
+            specification_pkgs = specification_pkgs + packages
```

Comment:
If `env.external_packages` is guaranteed to be a dict (even if empty), then we can just do:

```suggestion
    for package_type, packages in env.external_packages.items():
        specification_pkgs = specification_pkgs + packages
```